src/hera/workflows/v5/_mixins.py

```diff
@@ -170,6 +170,7 @@ def _build_env_from(self) -> Optional[EnvFromSource]:
 
 
 class _TemplateMixin(_BaseMixin):
+    name: str
```

Comment:
Mentioned on Sam's PR - [inline templates don't need a name](https://github.com/argoproj-labs/hera-workflows/pull/451#discussion_r1126197475) - up to you if you want to enforce naming within Hera code? The `Template` model class itself also has `name: Optional[str]`.